library/asn1write.c

```diff
@@ -354,6 +357,7 @@ int mbedtls_asn1_write_octet_string(unsigned char **p, const unsigned char *star
 }
 
 
+#if !defined(MBEDTLS_ASN1_PARSE_C)
```

Comment:
@daverodgman "replicated to avoid a dependency ASN1_WRITE_C on ASN1_PARSE_C" aren't we about to put such a dependency in with #8025?

Reply:
Just had a very quick look - doesn't look to me like that introduces a dependency? If it does, then I guess we can tidy things up and lose the conditional introduced here.